cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -495,7 +495,7 @@ class SnappyUnifiedMemoryManager private[memory](
       if (!enoughMemory) {
 
         // return immediately for OFF_HEAP with shouldEvict=false
-        if (offHeapNoEvict || tempManager) return false
+        if (offHeapNoEvict) return false
```

Comment:
Shouldn't the other call of "if (tempManager) { storagePool.acquireMemory(blockId, numBytes, numBytesToFree = 0)" also be removed to be same for both cases?